### ISSUE TITLE
JShint doesn't give an error for undefined variables

### ISSUE THREAD

**Issue by KhaledElAnsari** [AtomLinter/linter-jshint#323]
Hello

I just downloaded jshint using `apm install linter-jshint` and `npm install -g jshint` then I started testing it and it's awesome until I had a problem were I changed the name of a defined a variable and forgot to change it in some place but the JShint didn't notify me!

here's my code

```javascript
    var _IEsupport = typeof Date.prototype.toISOString === 'function';
    var now = new Date();
    var today;
    if(_IEcheck) {
      today = now.toISOString().substr(0,10);
      return today;
    }
    else {
       // ...
    }
```

**Comment by Arcanemagus** [comment]
Looks like you would need to enable the [`undef`](http://jshint.com/docs/options/#undef) rule in your JSHint configuration.
